### PR TITLE
Separate `@lexical/code` into more atomic modules

### PR DESCRIPTION
**Pull request by thegreatercurve**

### CODE REVIEW

File: packages/lexical-code/src/index.ts [hunk 1]
```diff
@@ -6,1206 +6,6 @@
...
-    editor.registerCommand(
-      MOVE_TO_START,
-      (payload): boolean => handleMoveTo(MOVE_TO_START, payload),
-      COMMAND_PRIORITY_LOW,
-    ),
-  );
-}
+export * from './CodeHighlighter';
+export * from './CodeHighlightNode';
+export * from './CodeNode';
```
Comment (zurfyx) [inline packages/lexical-code/src/index.ts:11]: Should we use named export rather than `*`? It did lead to accidentally exported modules in the past

**Review by zurfyx — APPROVED** [review]

File: packages/lexical-code/src/index.ts (hunk 1 above)
Comment (acywatson) [reply packages/lexical-code/src/index.ts:11]: Agree

**Review by acywatson — APPROVED** [review]
Fine with this, but curious what prompted it.

**Comment by thegreatercurve** [comment]
> Fine with this, but curious what prompted it.

A change that Lateef had made. This rebases some of his changes and updates them with recent changes to lexical/code

**Pull request merged**